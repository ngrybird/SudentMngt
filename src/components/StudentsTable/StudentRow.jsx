import React from "react";
import Ellipsis from "../../images/Ellipse.png";
class StudentRow extends React.Component {
  render() {
    return (
      <div className="table-row" data-testid="student-row">
        <span>
          <img src={Ellipsis} height="40px" width="40px" />
        </span>
        <span className="row-title">{this.props.name}</span>
        <span className="row-marks">{this.props.marks}</span>
      </div>
    );
  }
}
export default StudentRow;
