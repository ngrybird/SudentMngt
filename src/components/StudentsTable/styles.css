.table-container {
  width: 395px;
  height: auto;
  overflow-y: scroll;
  border: solid 0px;
  border-radius: 10px;
  margin-left: 40px;
  margin-top: 20px;
  margin-bottom: 20px;
  background: white;
  box-shadow: 0 8px 6px -6px grey;
}
.table-header {
  padding: 25px 30px;
  font-size: 14px;
  font-weight: 600;
  padding-bottom: 15px !important;
}
.table-header button {
  background: white;
  border: none;
  float: right;
  color: grey;
}
.table-header button:focus {
  border: none;
  outline: 0;
}
hr {
  color: #606060;
}
.student-list {
  height: 300px;
  overflow-y: auto;
}
.table-row {
  display: flex;
  position: relative;
  flex-flow: row wrap;
  align-items: center;
  margin-left: 10px;
  margin-top: 20px;
}
.row-title {
  font-size: 14px;
  color: grey;
  margin-left: 10px;
}
.row-marks {
  position: absolute;
  font-weight: 600;
  right: 20px;
}
