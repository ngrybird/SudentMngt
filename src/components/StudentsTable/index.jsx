import React from "react";
import StudentRow from "./StudentRow";
import "./styles.css";

class StudentsTable extends React.Component {
  constructor(props) {
    super(props);
    const descendingList = this.props.records.sort((f, s) => s.marks - f.marks);
    this.state = {
      listArrangement: "Descending",
      studentRecords: [...descendingList]
    };
  }
  handleSort = e => {
    console.log(e.target.innerText);
    const newState = {
      listArrangement:
        e.target.innerText === "Descending" ? "Ascending" : "Descending",
      studentRecords: this.state.studentRecords.reverse()
    };
    this.setState({ ...newState });
  };
  render() {
    console.log(this.state);
    return (
      <div className="table-container">
        <div className="table-header">
          <span>Students by average marks</span>
          <button onClick={this.handleSort}>
            {this.state.listArrangement}
          </button>
        </div>
        <hr />
        <div className="student-list">
          {this.state.studentRecords.map((records, i) => (
            <StudentRow key={i} {...records} />
          ))}
        </div>
      </div>
    );
  }
}

export default StudentsTable;
