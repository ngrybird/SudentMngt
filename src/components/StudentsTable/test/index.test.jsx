import React from "react";
import { render, fireEvent } from "@testing-library/react";
import StudentsTable from "../index";
import { studentsMarks } from "../../Dashboard/test/testData";

describe("<StudentsTable/>", () => {
  it("Verify Students table is rendered", () => {
    const { container, getByText, getAllByTestId } = render(
      <StudentsTable records={studentsMarks} />
    );
    const tableHeader = getByText("Students by average marks");
    const allRows = getAllByTestId("student-row");
    expect(allRows.length).toEqual(7);
    expect(tableHeader).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("simulate click to arrange list in ascending order", async () => {
    const { container, getByText, getAllByTestId } = render(
      <StudentsTable records={studentsMarks} />
    );

    fireEvent.click(getByText("Descending"));
    const allRows = getAllByTestId("student-row");

    // const newButton = getByText("Ascending");
    // expect(newButton).toBeInTheDocument();
  });
});
