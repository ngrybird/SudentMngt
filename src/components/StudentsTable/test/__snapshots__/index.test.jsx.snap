// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudentsTable/> Verify Students table is rendered 1`] = `
<div>
  <div
    class="table-container"
  >
    <div
      class="table-header"
    >
      <span>
        Students by average marks
      </span>
      <button>
        Descending
      </button>
    </div>
    <hr />
    <div
      class="student-list"
    >
      <div
        class="table-row"
        data-testid="student-row"
      >
        <span>
          <img
            height="40px"
            src="Ellipse.png"
            width="40px"
          />
        </span>
        <span
          class="row-title"
        >
          Annete Watson
        </span>
        <span
          class="row-marks"
        >
          9.3
        </span>
      </div>
      <div
        class="table-row"
        data-testid="student-row"
      >
        <span>
          <img
            height="40px"
            src="Ellipse.png"
            width="40px"
          />
        </span>
        <span
          class="row-title"
        >
          Calvin Steward
        </span>
        <span
          class="row-marks"
        >
          8.9
        </span>
      </div>
      <div
        class="table-row"
        data-testid="student-row"
      >
        <span>
          <img
            height="40px"
            src="Ellipse.png"
            width="40px"
          />
        </span>
        <span
          class="row-title"
        >
          Ralph Richards
        </span>
        <span
          class="row-marks"
        >
          8.7
        </span>
      </div>
      <div
        class="table-row"
        data-testid="student-row"
      >
        <span>
          <img
            height="40px"
            src="Ellipse.png"
            width="40px"
          />
        </span>
        <span
          class="row-title"
        >
          Bernard Murphy
        </span>
        <span
          class="row-marks"
        >
          8.2
        </span>
      </div>
      <div
        class="table-row"
        data-testid="student-row"
      >
        <span>
          <img
            height="40px"
            src="Ellipse.png"
            width="40px"
          />
        </span>
        <span
          class="row-title"
        >
          Arlene Robertson
        </span>
        <span
          class="row-marks"
        >
          8.2
        </span>
      </div>
      <div
        class="table-row"
        data-testid="student-row"
      >
        <span>
          <img
            height="40px"
            src="Ellipse.png"
            width="40px"
          />
        </span>
        <span
          class="row-title"
        >
          John Lane
        </span>
        <span
          class="row-marks"
        >
          8.1
        </span>
      </div>
      <div
        class="table-row"
        data-testid="student-row"
      >
        <span>
          <img
            height="40px"
            src="Ellipse.png"
            width="40px"
          />
        </span>
        <span
          class="row-title"
        >
          Pat McKinney
        </span>
        <span
          class="row-marks"
        >
          7.9
        </span>
      </div>
    </div>
  </div>
</div>
`;
